src/GPUActivityAgent.cpp

```diff
@@ -23,6 +23,14 @@
 #include "Waiter.hpp"
 #include "Environment.hpp"
 
+// IDLE SAMPLE COUNT of 10 is based upon a study of the idle behavior of CORAL-2
+// workloads of interest assuming the default 20ms sample rate (200ms idle).
+// We could use 200ms as the default for the agent, but this does not provide a
+// mechanism for user control of the idle period.  Using a count provides partial
+// user control in that the idleness period is defined by the requested agent
+// control loop time.
```

Comment:
This makes it sound like the goal is actually time-based rather than sample-based. Should `IDLE_SAMPLE_COUNT` be computed from `environment().period(M_WAIT_SEC)` and a `200ms` target instead?

Reply:
The ideal solution would be to provide a control for both the control loop time (via waiter) and the idle time until the agent takes action.  We don't currently have the agent built with this in mind, so having the idle time be a function of the user input we do have seemed preferable as there could be a workload that has the behavior
... N milliseconds active, 200 ms idle, N milliseconds active ...
We can either:
1. Make it sample based, giving the user some control
2. Make it time based, and risk a pathological case described above
3. Add the idle time/sample count to the policy

Of these I like either 1 or 2

Reply:
Why a ```#define``` instead of a const like M_CPU_ACTIVITY_CUTOFF and set in initializer list?

If we expect this is something the user would modify, it should be a policy parameter.